src/runtime/gecko/l20n-html.js

```diff
@@ -0,0 +1,56 @@
+import { prioritizeLocales } from '../../intl/locale';
+import { contexts, Localization } from '../../bindings/html';
+import { documentReady, getResourceLinks, getMeta } from './util';
+import { keysFromContext, valueFromContext } from '../../lib/format';
+
+Components.utils.import('resource://gre/modules/Services.jsm');
+Components.utils.import('resource://gre/modules/L20n.jsm');
+Components.utils.import('resource://gre/modules/IntlMessageContext.jsm');
+
+const functions = {
+  OS: function() {
+    switch (Services.appinfo.OS) {
+      case 'WINNT':
+        return 'win';
+      case 'Linux':
+        return 'lin';
+      case 'Darwin':
+        return 'mac';
+      case 'Android':
+        return 'android';
+      default:
+        return 'other';
+    }
+  }
+};
+
+function requestBundles(requestedLangs = navigator.languages) {
+  return documentReady().then(() => {
+    const { defaultLang, availableLangs } = getMeta(document.head);
+    const resIds = getResourceLinks(document.head);
+
+    const newLangs = prioritizeLocales(
+      defaultLang, Object.keys(availableLangs), requestedLangs
+    );
+
+    return newLangs.map(
+      lang => new ResourceBundle(lang, resIds)
+    );
+  });
+}
+
+function createContext(lang) {
+  return new MessageContext(lang, { functions });
+}
+
+document.l10n = new Localization(document, requestBundles, createContext);
```

Comment:
I think I preferred an approach in which `MessageContext` is exposed on a global namespace, like `Intl`, rather than having to pass `createContext` around.  What where the reasons for not doing it?